libraries/tock-register-interface/src/macros.rs

```diff
@@ -325,6 +325,7 @@ macro_rules! test_fields {
     };
 }
 
+#[cfg(feature = "std_unit_tests")]
```

Comment:
Would it work by integrating this `cfg` in the `cfg(test)` inside of the macro? Or not because the features would be those of the calling crate?

Reply:
I tried that and it didn't work. From my tests and a bit of googling, it appears that the conditional compilation with config has to be at whole-macro level.

Reply:
Yes that makes sense. Since the macro's content can be expanded anywhere the feature wouldn't exist outside of this crate. This part looks good to me then.